projects/ccf-rui/src/app/modules/left-sidebar/left-sidebar.component.ts

```diff
@@ -72,4 +78,24 @@ export class LeftSidebarComponent {
   setSideFromLabel(label: 'L' | 'R'): void {
     this.model.setSide(label === 'L' ? 'left' : 'right');
   }
+
```

Comment:
This business logic should really be handled in the state layer

Reply:
This isn't my code do you still want me to address it in this pr?

Reply:
Sorry if it wasn't clear, I meant the togglePreviousRegistrationBlocks function below.

Reply:
@axdanbol can give feedback regarding the best way to do that with NGXS

Reply:
This function can pretty much simply be moved to the ModelState. A few things to keep in mind though is to NOT add `@DataAction()` to the function because it is calling other actions! You probably want to return the array of previous vis items since you won't have access to the component. Also the clone on line 94 is unnecessary since map creates a new array.

LeftSidebarComponent
```
togglePreviousRegistrationBlocks(visible: boolean): void {
  this.previousVisibilityItems = this.model.toggleVisibility(visible, this.previousVisibilityItems);
}
```

ModelState
```
toggleVisibility(visible: boolean, previous: ItemType[]): ItemType[] {
  ...
}